### PR TITLE
[R4ML-{74,237}] Check and Fix Cox Predict Potential mismatch with R a…

### PR DESCRIPTION
**Pull request by aloknsingh**
…nd accuracy test

1) accuracy test for predict added.
2) fix T_Y as per sasha's suggestion
3) beautify the code.
4) info added that predict output will be always sorted wrt to TS

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by aloknsingh** [comment]
retest this please

**Comment by aloknsingh** [comment]
retest this please

**Comment by aloknsingh** [comment]
retest this please

File: R4ML/tests/testthat/test-ml.model.cox.R [hunk 1]
```diff
@@ -40,61 +71,129 @@ test_that("r4ml.coxph", {
...
+  # test the final prediction
+  
+  # r4ml prediction
   h_predict <- predict(h_cox, data = hf)
-  r_predict <- predict(r_cox, data = df)
+  h_predict_df <- SparkR::as.data.frame(h_predict) # r frame
+  
+  # since R's cox predict need to be called many times,
+  # we have the wrapper for this
+  r_predict <- r_cox_predict(r_cox, df)
```
Comment (bdwyer2) [inline R4ML/tests/testthat/test-ml.model.cox.R]: Why did you change `predict(` to `r_cox_predict(`? It seems to have broken this.
